src/components/token/Tokens.js

```diff
@@ -24,18 +27,101 @@ class Tokens extends React.Component {
             tokens: [],
             hasAfter: false,
             hasBefore: false,
-            searchText: ""
+            searchText: "",
+            sortBy: "uid",
+            order: "asc",
+            page: 1,
+            //This variable stores the pages and its searchAfter params that is passed to explorer-service
+            pageSearchAfter: [],
+            loading: true
         }
     }
 
-    onSearchButtonClicked = () => {
-        // TODO: Implement behavior when search button is clicked
+    componentDidMount = async() => {
+        //"Click" on search to make the first query
+        await this.onSearchButtonClicked();
+        this.setState({loading: false})
+    }
+
+    resetPageSearchAfter = () => {
+        this.setState({pageSearchAfter: [
```

Comment:
What's the expected format of elements in `pageSearchAfter` array?

Reply:
pageSearchAfter calculates the searchAfter param that needs to be passed to explorer-service in order to get the next/previous page. We use this array to store already-calculated values, so we do not need to recalculate them if user is requesting an already-navigated page.

Reply:
I feel this could be added to the state docstring so it's clear in the code

Reply:
I've added the state docstring